### PR TITLE
[socket-io-client] update to 3.0.0 

### LINKED ISSUE #18641: [socket-io-client] update to 3.0.0
Library name: socket-io-client

New version number: 3.0.0

Related issue: #18603

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #18641 

Update socket-io-client to the latest version 

Note: no feature need to test

### CODE REVIEW

File: ports/socket-io-client/fix-error-C3321.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/internal/sio_client_impl.cpp b/src/internal/sio_client_impl.cpp
+index f78aea4..6780e26 100644
+--- a/src/internal/sio_client_impl.cpp
++++ b/src/internal/sio_client_impl.cpp
+@@ -286,7 +286,7 @@ namespace sio
+         if(ec || m_con.expired())
+         {
+             if (ec != asio::error::operation_aborted)
+-                LOG("ping exit,con is expired?"<<m_con.expired()<<",ec:"<<ec.message()<<endl){};
++                LOG("ping exit,con is expired?"<<m_con.expired()<<",ec:"<<ec.message()<<endl);
```
Comment (JonLiu1993) [inline ports/socket-io-client/fix-error-C3321.patch:10]: This patch fix :error C3321: an initializer list is unexpected in this context

File: ports/socket-io-client/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "socket-io-client",
+  "version": "3.0.0",
+  "description": "C++11 implementation of Socket.IO client",
+  "homepage": "https://github.com/socketio/socket.io-client-cpp",
+  "dependencies": [
+    "boost",
```
Comment (ysc3839) [inline ports/socket-io-client/vcpkg.json:7]: I think boost is not required, because it uses standalone asio.

Comment (JonLiu1993) [reply ports/socket-io-client/vcpkg.json:7]: @ysc3839, sorry to reply you so late,I investigated the previous version that depends on boost, the latest version has removed the dependency on boost, I will make changes, thank you.

**Pull request merged**